blueprint/core/blueprint_EcmascriptEngine.cpp

```diff
@@ -252,8 +252,101 @@ namespace blueprint
+        {
+            ~TimeoutFunctionManager() override {
+                for(const auto &[id, timer] : timeoutFunctions)
+                    stopTimer(id);
+            }
+
+            juce::var clearTimeout(const int id)
+            {
+                stopTimer(id);
+                const auto f = timeoutFunctions.find(id);
+                if(f != timeoutFunctions.cend())
+                    timeoutFunctions.erase(f);
+                return juce::var();
+            }
+
+            int newTimeout(const juce::var::NativeFunction f, const int timeoutMillis, const std::vector<juce::var>&& args, const bool repeats=false)
+            {
+                static int nextId = 0;
+                timeoutFunctions.emplace(nextId, TimeoutFunction(f, std::move(args), repeats));
+                startTimer(nextId, timeoutMillis);
+                return nextId++;
+            }
+
+            void timerCallback(int id) override
+            {
+                const auto f = timeoutFunctions.find(id);
+                if(f != timeoutFunctions.cend())
+                {
+                    const auto cb = f->second;
+                    std::invoke(cb.f, juce::var::NativeFunctionArgs(juce::var(), cb.args.data(), static_cast<int>(cb.args.size())));
+                    if(!cb.repeats)
+                    {
+                        stopTimer(id);
+                        timeoutFunctions.erase(f);
+                    }
+                }
+            }
+
+            private:
+                struct TimeoutFunction
+                {
+                    TimeoutFunction(const juce::var::NativeFunction _f, const std::vector<juce::var> &&_args, const bool _repeats=false)
+                    : f(_f), args(std::move(_args)), repeats(_repeats) {}
+
+                    const juce::var::NativeFunction f;
+                    std::vector<juce::var> args;
+                    const bool repeats;
+                };
+
+                std::map<int, TimeoutFunction> timeoutFunctions;
+        };
+
+        // IsSetter is true for setTimeout / setInterval
+        // and false for clearTimeout / clearInterval
+        template <bool IsSetter = false, bool Repeats = false, typename MethodType>
+        void registerNativeTimerFunction(const char* name, MethodType method)
+        {
+            registerNativeProperty(name, juce::var::NativeFunction([this, name, method] (const juce::var::NativeFunctionArgs& _args) -> juce::var {
+                if constexpr (IsSetter)
+                {
+                    if(_args.numArguments < 2 || !_args.arguments[0].isMethod() || !_args.arguments[1].isDouble())
+                        throw Error(juce::String(name) + " requires a callback and time in milliseconds");
+                    // build a vector holding all additional arguments
+                    std::vector<juce::var> args(_args.numArguments - 2);
+                    for(auto i=2; i<_args.numArguments; i++)
```

Comment:
Can we use the std vector constructor `std::vector<juce::var> args (_args.arguments + 2, _args.arguments + args.numArguments)`? I don't know if the raw pointer there counts as an iterator type, so maybe not, but that seems like it'd be a tiny cleaner if we can

Reply:
Ooh, I'll see if there is a slicker initialization yeah

Reply:
Yeaaaah, pointers as iterators, that works exactly as you suggested!